proposals/push-constants.md

```diff
@@ -0,0 +1,73 @@
+# PushConstants
```

Comment:
Should we really use the term 'push constants' which seems like a non-universal, backend specific name as opposed to something like 'immediate data'?

Reply:
Done!